juju/terraform-provider-juju
# Base for machine resource

## Description

There are two parts. One separate the concerns of the terraform provider from the internal juju code which uses the juju api. 

Two, add Base to the schema for machines, while deprecating Series. Regardless of which is supplied by the user, a base will be provided to the AddMachine api call to juju. 

A string validator to cover Bases is also added, its used by machine and will be used for application as well.

Part of the effort to replace Series with Base within the terraform plans for juju as seen in #249

## Type of change

- Change existing resource

## QA steps

Here is a plan to test the change. Please set manual machines as well, to ensure no changes there.

```tf
provider juju {}

resource "juju_model" "testmodel" {
  name = "machinetest"
}

resource "juju_machine" "my-machine" {
  model = juju_model.testmodel.name
  base = "ubuntu@22.04"
}
```
